x/savings/keeper/deposit_test.go

```diff
@@ -0,0 +1,149 @@
+		depositor                 sdk.AccAddress
+		amount                    sdk.Coins
+		numberDeposits            int
+		expectedAccountBalance    sdk.Coins
+		expectedModAccountBalance sdk.Coins
+		expectedDepositCoins      sdk.Coins
+	}
+	type errArgs struct {
+		expectPass bool
+		contains   string
+	}
+	type depositTest struct {
+		name    string
+		args    args
+		errArgs errArgs
+	}
+	testCases := []depositTest{
+		{
+			"valid",
+			args{
+				depositor:                 sdk.AccAddress(crypto.AddressHash([]byte("test"))),
+				amount:                    sdk.NewCoins(sdk.NewCoin("bnb", sdk.NewInt(100))),
+				numberDeposits:            1,
+				expectedAccountBalance:    sdk.NewCoins(sdk.NewCoin("bnb", sdk.NewInt(900)), sdk.NewCoin("btcb", sdk.NewInt(1000))),
+				expectedModAccountBalance: sdk.NewCoins(sdk.NewCoin("bnb", sdk.NewInt(100))),
+				expectedDepositCoins:      sdk.NewCoins(sdk.NewCoin("bnb", sdk.NewInt(100))),
+			},
+			errArgs{
+				expectPass: true,
+				contains:   "",
+			},
+		},
+		{
+			"valid multi deposit",
+			args{
+				depositor:                 sdk.AccAddress(crypto.AddressHash([]byte("test"))),
+				amount:                    sdk.NewCoins(sdk.NewCoin("bnb", sdk.NewInt(100))),
+				numberDeposits:            2,
+				expectedAccountBalance:    sdk.NewCoins(sdk.NewCoin("bnb", sdk.NewInt(800)), sdk.NewCoin("btcb", sdk.NewInt(1000))),
+				expectedModAccountBalance: sdk.NewCoins(sdk.NewCoin("bnb", sdk.NewInt(200))),
+				expectedDepositCoins:      sdk.NewCoins(sdk.NewCoin("bnb", sdk.NewInt(200))),
+			},
+			errArgs{
+				expectPass: true,
+				contains:   "",
+			},
+		},
+		{
+			"invalid deposit denom",
+			args{
+				depositor:                 sdk.AccAddress(crypto.AddressHash([]byte("test"))),
+				amount:                    sdk.NewCoins(sdk.NewCoin("fake", sdk.NewInt(100))),
+				numberDeposits:            1,
+				expectedAccountBalance:    sdk.Coins{},
+				expectedModAccountBalance: sdk.Coins{},
+				expectedDepositCoins:      sdk.Coins{},
+			},
+			errArgs{
+				expectPass: false,
+				contains:   "invalid deposit denom",
+			},
+		},
+		{
+			"insufficient funds",
+			args{
+				depositor:                 sdk.AccAddress(crypto.AddressHash([]byte("test"))),
+				amount:                    sdk.NewCoins(sdk.NewCoin("bnb", sdk.NewInt(10000))),
+				numberDeposits:            1,
+				expectedAccountBalance:    sdk.Coins{},
+				expectedModAccountBalance: sdk.Coins{},
+				expectedDepositCoins:      sdk.Coins{},
+			},
+			errArgs{
+				expectPass: false,
+				contains:   "insufficient funds",
+			},
+		},
+	}
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			// create new app with one funded account
+
+			// Initialize test app and set context
+			tApp := app.NewTestApp()
+			ctx := tApp.NewContext(true, tmproto.Header{Height: 1, Time: tmtime.Now()})
+			authGS := app.NewFundedGenStateWithCoins(
+				tApp.AppCodec(),
+				[]sdk.Coins{
+					sdk.NewCoins(
+						sdk.NewCoin("bnb", sdk.NewInt(1000)),
```

Comment:
I would set the initial depositor balance as part of the `args` ie `tc.args.initialDepositorBalance`

Reply:
Updated in latest